Apns cred check

viktormuller opened the pull request:
Checks that iOS app provisioning profile and credential is aligned.

This is the no.1 problem developer's run into hence an automated check should greatly help debugging. Needs corresponding changes in iOS apps as well.

dprothero committed:
Merge pull request #3 from TwilioDevEd/DEVED-1386

Refactor config.json to config.js

eddiezane committed:
Remove placeholder values from README.md

viktormuller committed:
Added x-www-form-urlencoded parameter support

This is required as the Android quickstart app is using x-www-form-urlencoded parameters.

kwhinnery committed:
Merge pull request #4 from viktormuller/patch-1

Added x-www-form-urlencoded parameter support

Commit:
Fixes helper library change of notifications to notify

Commit:
Removed config from config.js

Commit:
Merge branch 'master' of https://github.com/TwilioDevEd/notifications-quickstart-node

Commit:
Added config info to config.js

Commit:
removed config.js from gitignore

Commit:
Adds support for Facebook Messenger guide.

Commit:
Merge remote-tracking branch 'deved/master' into apns-cred-check

Merged refactoring from deved/master for createBinding function.

Commit:
Checks that iOS app provisioning profile and credential are aligned.

kwhinnery commented on index.js:
```diff
@@ -24,13 +24,42 @@ app.get('/', function(request, response) {
+  if (bindingType === "apn"){
+    var apnCredMatched = false;
+    service.fetch().then(function(serviceObj){
+      client.notify.v1.credentials(serviceObj.apnCredentialSid).fetch()
```
return this promise, and move the `then` just before your final `catch` below - then there will be only one `catch` for this block.

Commit:
Removes credential config and improves APNS credential check

Removed APN and GCM credential SID configs as configuration is now
available via the console. Therefore also removed create_service.js.

Furthermore improved promise usage in index.js when checking APNS
credential alignment.

viktormuller commented:
@kwhinnery Thanks. I have tried to improve it. Also simplified the config by removing the apn and gcm credential SIDs as those are not used anymore. They are configured via the console and they can just cause confusion as they can be out of sync with actual config in console.

Commit:
Fixes diagnostic index page

Commit:
Better error response

Commit:
Merge remote-tracking branch 'deved/master' into apns-cred-check

Commit:
Adds support for optional Endpoint parameter.

viktormuller commented:
@coderkat This also supports now endpoint ID autogeneration by accepting registration requests without the Endpoint ID and including Endpoint ID in the response.

coderkat-zz commented on index.js:
```diff
@@ -14,47 +15,95 @@ app.use(express.static('public'));
 // correctly
 app.get('/', function(request, response) {
-  response.render('index.jade', {
+
```
Nitpick, but we don't need this blank line

coderkat-zz commented on index.js:
```diff
@@ -14,47 +15,95 @@ app.use(express.static('public'));
     TWILIO_ACCOUNT_SID: env.TWILIO_ACCOUNT_SID,
     TWILIO_AUTH_TOKEN: env.TWILIO_AUTH_TOKEN,
     TWILIO_NOTIFICATION_SERVICE_SID: env.TWILIO_NOTIFICATION_SERVICE_SID,
+
```
Nitpick: blank line

coderkat-zz commented on index.js:
```diff
@@ -14,47 +15,95 @@ app.use(express.static('public'));
     TWILIO_NOTIFICATION_SERVICE_SID: env.TWILIO_NOTIFICATION_SERVICE_SID,
+
   });
+
```
Nitpick: blank line

coderkat-zz reviewed (commented):
Looks good! Just a few very nitpicky style comments.

viktormuller commented:
@coderkat Thx. Done. Sorry for the delay have not seen the comments.
